src/kv/apply_changes.h

```diff
@@ -70,7 +71,15 @@ namespace kv
     }
 
     bool ok = true;
-    if (has_writes)
+
+    if (expected_rollback_count.has_value())
+    {
+      // Assumes that all maps in the transaction point to the same store
+      auto store = changes.begin()->second.map->get_store();
+      ok = store->check_rollback_count(expected_rollback_count.value());
+    }
+
+    if (ok && has_writes)
```

Comment:
I think the loop below, on line 94, can also be skipped entirely `if (!ok)`. All it is doing is checking _another_ reason that `ok` might become `false`. (This was true previously, but I just noticed it now).

Reply:
Yes